### PR TITLE
Added AuthenticationService, JwtAuthenticationFilter, JwtUtil, Authen…

### PR DESCRIPTION
**Pull request by Nikname2303**
…ticationController

### CODE REVIEW

File: src/main/java/com/example/bookshop/controller/AuthenticationController.java [hunk 1]
```diff
@@ -16,10 +19,16 @@
...
+    private final AuthenticationService authenticationService;
 
     @PostMapping("/registration")
     public UserResponseDto register(@RequestBody @Valid UserRegistrationRequestDto request)
             throws RegistrationException {
         return userService.register(request);
     }
+
+    @PostMapping("/login")
+    public UserLoginResponseDto login(@RequestBody UserLoginRequestDto request) {
```
Comment (andrii-hoienko) [inline src/main/java/com/example/bookshop/controller/AuthenticationController.java]:
```suggestion
    public UserLoginResponseDto login(@RequestBody @Valid UserLoginRequestDto request) {
```

File: src/main/java/com/example/bookshop/security/JwtAuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+                    userDetails, null, userDetails.getAuthorities()
+            );
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer ")) {
```
Comment (andrii-hoienko) [inline src/main/java/com/example/bookshop/security/JwtAuthenticationFilter.java]: Move to constants

File: src/main/java/com/example/bookshop/security/JwtAuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            return bearerToken.substring(7);
```
Comment (andrii-hoienko) [inline src/main/java/com/example/bookshop/security/JwtAuthenticationFilter.java]: Magic number

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
Nice, left some minor comments

**Review by vovasalyha — APPROVED** [review]

**Pull request merged**